# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/webgpu/BindGroupLayout.cpp
# Commit: dfd34fb1aade
# Full Hash: dfd34fb1aade1295579715b1c21f1e51bbfe370e
# Author: Dzmitry Malyshau <dmalyshau@mozilla.com>
# Date: 2021-05-18 21:36:28
# Regressor Bug: 1709462
# File Overlap Count: 2
# Description:
#   Bug 1709462 - Fix freeing of WebGPU bind group layouts r=jimb
#   
#   When the pipeline layout is implicit at pipeline creation,
#   we collect the IDs of BGLs, to be able to produce GPUBindGroupLayout
#   object upon user request. However, the produced object didn't know if
# ==============================================================================

diff -r 5c6f0950714d -r dfd34fb1aade dom/webgpu/BindGroupLayout.cpp
--- a/dom/webgpu/BindGroupLayout.cpp	Tue May 18 16:24:49 2021 +0000
+++ b/dom/webgpu/BindGroupLayout.cpp	Tue May 18 16:26:27 2021 +0000
@@ -15,8 +15,8 @@
 GPU_IMPL_CYCLE_COLLECTION(BindGroupLayout, mParent)
 GPU_IMPL_JS_WRAP(BindGroupLayout)
 
-BindGroupLayout::BindGroupLayout(Device* const aParent, RawId aId)
-    : ChildOf(aParent), mId(aId) {}
+BindGroupLayout::BindGroupLayout(Device* const aParent, RawId aId, bool aOwning)
+    : ChildOf(aParent), mId(aId), mOwning(aOwning) {}
 
 BindGroupLayout::~BindGroupLayout() { Cleanup(); }
 
@@ -24,7 +24,7 @@
   if (mValid && mParent) {
     mValid = false;
     auto bridge = mParent->GetBridge();
-    if (bridge && bridge->IsOpen()) {
+    if (mOwning && bridge && bridge->IsOpen()) {
       bridge->SendBindGroupLayoutDestroy(mId);
     }
   }